app/src/main/java/com/android/sample/ui/profile/Profile.kt

```diff
@@ -321,8 +335,56 @@ fun ActivityRow(activity: Activity, onClickAction: () -> Unit, testTag: String)
               color = Color.Black)
           Text(text = activity.description, fontSize = SUBTITLE_FONTSIZE.sp, color = Color.Gray)
         }
+        if (category == "past") {
+          ReviewActivityButtons(activity.likes[userId]) { review ->
+            listActivitiesViewModel.reviewActivity(activity, userId, review)
+          }
+        }
       }
 }
+
+@Composable
+fun ReviewActivityButtons(currentReview: Boolean?, review: (Boolean?) -> Unit) {
+  var isLiked: Boolean? by remember { mutableStateOf(currentReview) }
+  Row {
+    IconButton(
+        onClick = {
+          isLiked = if (isLiked == true) null else true
+          review(isLiked)
+        },
+        colors =
+            IconButtonDefaults.iconButtonColors(
+                containerColor = if (isLiked == true) Color(0xFF048531) else Color.Transparent,
+                contentColor =
+                    if (isLiked == true) MaterialTheme.colorScheme.onError
+                    else MaterialTheme.colorScheme.onSurface),
+        modifier = Modifier.testTag("likeIconButton_${isLiked == true}")) {
+          Icon(imageVector = Icons.Default.ThumbUp, contentDescription = "Like")
+        }
+    Spacer(modifier = Modifier.width(MEDIUM_PADDING.dp))
+    IconButton(
+        onClick = {
+          isLiked = if (isLiked == false) null else false
```

Comment:
> Checking that isLiked == true is necessary because it allows us to manage the null case. Try it in your local environment to verify ! :) I took some time to think about this line because it seemed very odd to me. But I couldn't find something better.

Check this comment